pkg/rpc/client.go

```diff
@@ -149,47 +158,61 @@ func (c *Client) GetVersion(ctx context.Context) (string, error) {
 	var resp response[struct {
 		Version string `json:"solana-core"`
 	}]
-	if err := c.getResponse(ctx, "getVersion", []interface{}{}, &resp); err != nil {
+	if err := c.getResponse(ctx, "getVersion", []any{}, &resp); err != nil {
 		return "", err
 	}
 	return resp.Result.Version, nil
 }
 
 func (c *Client) GetSlot(ctx context.Context) (int64, error) {
 	var resp response[int64]
-	if err := c.getResponse(ctx, "getSlot", []interface{}{}, &resp); err != nil {
+	if err := c.getResponse(ctx, "getSlot", []any{}, &resp); err != nil {
 		return 0, err
 	}
 	return resp.Result, nil
 }
 
-func (c *Client) GetBlockProduction(ctx context.Context, firstSlot *int64, lastSlot *int64) (*BlockProduction, error) {
+func (c *Client) GetBlockProduction(
+	ctx context.Context, identity *string, firstSlot *int64, lastSlot *int64,
+) (*BlockProduction, error) {
+	// can't provide a last slot without a first:
+	if firstSlot == nil && lastSlot != nil {
+		panic("can't provide a last slot without a first!")
+	}
+
 	// format params:
-	params := make([]interface{}, 1)
+	config := make(map[string]any)
+	if identity != nil {
+		config["identity"] = *identity
+	}
 	if firstSlot != nil {
-		params[0] = map[string]interface{}{"range": blockProductionRange{FirstSlot: *firstSlot, LastSlot: lastSlot}}
+		blockRange := map[string]int64{"firstSlot": *firstSlot}
+		if lastSlot != nil {
+			// make sure first and last slot are in order:
+			if *firstSlot > *lastSlot {
+				panic(fmt.Errorf("last slot %v is greater than first slot %v", *lastSlot, *firstSlot))
```

Comment:
Can we just return this as an error instead of panicking? Less panicking is good.